mlir/include/mlir/Target/MIOpenCPP.h

```diff
@@ -221,21 +228,24 @@ class PopulateParamsBase {
   }
 
   static void
-  obtainInputVecLen(llvm::StringMap<std::pair<size_t, int64_t>> &dimIndexVal,
+  obtainInputVecLen(ConvolutionContext &ctx,
                     int64_t &vecLen) {
+    auto dimIndexVal = ctx.dimIndexVal;
     if (dimIndexVal["ni"].first == 3) {
       vecLen = dimIndexVal["ni"].second;
     } else if (dimIndexVal["ci"].first == 3) {
       vecLen = dimIndexVal["ci"].second;
     } else {
-      // Not vectorizable
-      // TODO(optimize): For 1x1, stride 1, pad 0 conv, vecLen is hi * wi
-      vecLen = 1;
+      if(ctx.strideVal[0] == 1 && ctx.strideVal[1] == 1 && ctx.paddingVal[0] == 0 && ctx.paddingVal[1] == 0)
```

Comment:
We need to also check `paddingVal[2]` and `paddingVal[3]`?

Reply:
What are mappings from ctx.paddingVal[4] to leftpadH/W and rightPadH/W?